virttest/utils_test/__init__.py

```diff
@@ -2200,6 +2226,31 @@ def unload_stress(stress_type, params, vms=None, remote_server=False):
                    remote_server=remote_server).unload_stress()
 
 
+def prepare_profile(test, fpath, pat, repl):
+    """
+    This is to prepare client profile to be run on client.
+    :param test: kvm test object
+    :param fpath: profile to be run
+    :param pat: pattern to be replaced
+    :param repl: replacement strings includes server_ip, duration, threads etc.
+    :return: no explicit return. But profile on fpath would be edited and ready to be run on client guest
+    :raise: TestError: raised if unable to edit the given profile
+    """
+    try:
+        with open(fpath, 'r+') as profile_content:
+            tempstr = profile_content.read()
+            profile_content.truncate(0)
+            pat_repl = zip(pat, repl)
+            logging.debug("In prepare profile: pattren : %s", pat)
+            logging.debug("In prepare profile: replace with : %s", repl)
+            for pattern, replace in pat_repl:
+                tempstr = tempstr.replace(pattern, replace)
+            profile_content.write(tempstr)
+        logging.debug("Profile xml to be run : %s ", tempstr)
```

Comment:
`profile_content.close()` is required here.

Reply:
I think `with` should take care of automatic file closure.

Reply:
got it :)